tests/Builder/Stage/ProjectStageTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Stage;
+
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+use function MongoDB\object;
+
+/**
+ * Test $project stage
+ */
+class ProjectStageTest extends PipelineTestCase
+{
+    public function testConditionallyExcludeFields(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::project(
+                // Array unpacking is necessary for name that are not valid argument names,
+                // in PHP, they must be defined before named arguments.
+                ...[
+                    'author.first' => 1,
+                    'author.last' => 1,
+                    'author.middle' => Expression::cond(
+                        if: Expression::eq(
+                            '',
+                            Expression::stringFieldPath('author.middle'),
+                        ),
+                        then: Expression::variable('REMOVE'),
+                        else: Expression::stringFieldPath('author.middle'),
+                    ),
+                ],
+                title: 1,
+            ),
+        );
+
+        $this->assertSamePipeline(Pipelines::ProjectConditionallyExcludeFields, $pipeline);
+    }
+
+    public function testExcludeFieldsFromEmbeddedDocuments(): void
+    {
+        $pipeline = new Pipeline(
+            // Both stages are equivalents
+            Stage::project(
+                ...['author.first' => 0],
+                ...['lastModified' => 0],
```

Comment:
How are you using the splat operator twice, once for each positional argument? Wouldn't the correct syntax be: `...['author.first' => 0, 'lastModified' => 0]`?

Reply:
Both work. I used this syntax to keep 1 field per line.

Reply:
TIL. There is seemingly no mention of this within PHP docs (just read through [Function Arguments](https://www.php.net/manual/en/functions.arguments.php)). Tested it myself: https://3v4l.org/sJ5KB#v8.3.2